src/main/scala/is/hail/distributedmatrix/BlockMatrix.scala

```diff
@@ -59,44 +59,64 @@ object BlockMatrix {
   def from(irm: IndexedRowMatrix, blockSize: Int): M =
     irm.toHailBlockMatrix(blockSize)
 
+  def random(hc: HailContext, rows: Int, cols: Int, blockSize: Int = defaultBlockSize): M = {
+    val part = GridPartitioner(blockSize, rows, cols)
+    new BlockMatrix(
+      new RDD[((Int, Int), BDM[Double])](hc.sc, Nil) {
+        override val partitioner = Some(part)
+
+        def getPartitions: Array[Partition] = Array.tabulate(part.numPartitions)(i =>
+          new Partition {
+            def index: Int = i
+          })
+
+        def compute(split: Partition, context: TaskContext): Iterator[((Int, Int), BDM[Double])] = {
+          val (i, j) = part.blockCoordinates(split.index)
+          // FIXME seed so reproducible
+          Iterator(((i, j), BDM.rand[Double](part.blockRowNRows(i), part.blockColNColumns(j))))
+        }
+      },
+      blockSize, rows, cols)
+  }
```

Comment:
should you set the partitioner to be the GridPartitioner?

Reply:
I do above: `override val partitioner = Some(part)`.